extensions/crypto/01_crypto.js

```diff
@@ -97,6 +196,239 @@
 
       return result.buffer;
     }
+
+    async sign(algorithm, key, data) {
+      const prefix = "Failed to execute 'sign' on 'SubtleCrypto'";
+
+      webidl.assertBranded(this, SubtleCrypto);
+      webidl.requiredArguments(arguments.length, 3);
+
```

Comment:
WebIDL converters always must run before all other algorithm steps. They can not be intermingled.

Reply:
I'll do this for `sign` but for `generateKey` WPT expects us to throw error for algorithm normalization before running converters on other arguments. eg: 
```typescript
// should throw for wrong hash identifier instead of extratable !== boolean
generateKey({hash: "INVALID", name:"HMAC"}, 0, ["sign"]);
```

Reply:
What I meant was that the webidl converters have to run _before_ the `Let normalizedAlgorithm be the result of normalizing an algorithm`.

So the code should look like this:
```js
class SubtleCrypto {
    async sign(algorithm, key, data) {
      webidl.assertBranded(this, SubtleCrypto);
      webidl.requiredArguments(arguments.length, 3);
      algorithm = webidl.converters.AlgorithmIdentifier(algorithm, {
        prefix,
        context: "Argument 1",
      });
      key = webidl.converters.CryptoKey(key, {
        prefix,
        context: "Argument 2",
      });
      data = webidl.converters.BufferSource(data, {
        prefix,
        context: "Argument 3",
      });
      const normalizedAlgorithm = normalizeAlgorithm(algorithm, "sign");

      // ...
    }
}
```

WebIDL converters need to all be finished before the operation steps defined in the spec.

Reply:
Ah! I misunderstood, will do. Also since `CryptoKey` is an interface it won't have a converter?

Reply:
done